packages/cells/src/widget.ts

```diff
@@ -492,6 +495,30 @@ export class Cell<T extends ICellModel = ICellModel> extends Widget {
     this.inputHidden = !!jupyter.source_hidden;
   }
 
+  /**
+   * Load the tag state from the model.
+   */
+  loadTagsState(): void {
+    const tags = (this.model.getMetadata('tags') as string[]) ?? [];
+
+    // Remove all existing tag data attributes
+    const attributes = Array.from(this.node.attributes) as Attr[];
+    for (const attr of attributes) {
+      if (attr.name && attr.name.startsWith('data-tag-')) {
+        this.node.removeAttribute(attr.name);
+      }
+    }
+
+    // Add data attributes for current tags
+    for (const tag of tags) {
+      if (typeof tag === 'string' && tag.length > 0) {
+        // Sanitize tag name for use as data attribute
+        const sanitizedTag = tag.replace(/[^a-zA-Z0-9-_]/g, '-');
+        this.node.setAttribute(`data-tag-${sanitizedTag}`, 'true');
```

Comment:
Do we need `true` here or can we create data attributes without value instead?

```suggestion
        this.node.setAttribute(`data-tag-${sanitizedTag}`, '');
```

https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute#:~:text=You%20should%20set%20value%20to,necessarily%20do%20what%20you%20expect.